homeassistant/components/ccl/__init__.py

```diff
@@ -0,0 +1,34 @@
+"""The CCL Electronics integration."""
+
+from __future__ import annotations
+
+from aioccl import CCLDevice, CCLServer
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a config entry for a single CCL device."""
+    entry.runtime_data = CCLDevice(entry.data["passkey"])
+
+    CCLServer.add_copy(entry.runtime_data)
+    await CCLServer.run()
```

Comment:
you are launching your own webserver here which is not something we want. There is a built in `webhook` component that can be used to allow push style HTTP callbacks, take a look at `tedee` for how those are registered. Also, is there any way you can push the instance details to CCL, instead of the user having to go to the app to configure that manually?

Reply:
Thanks for the advice. It is now changed to listen for requests using `webhook `api.
Here is the test command I input to the zsh terminal in dev container (about half of them are binary sensors which are not included in this PR):
`curl localhost:8123/api/webhook/c2507495 -X POST -H 'Content-Type: application/json' -d '{"serial_no": "1061616322","mac_address": "48:31:B7:06:D5:59","model":"HA100","fw_ver":"v1.00","datetime":"2024-12-26+02:47:46","abar":1014.1,"rbar":1012.9,"t1dew":16.1,"t1feels":22.9,"t1chill":22.9,"inhum":63,"intem":23.0,"t1solrad":0,"t1hum":66,"t1tem":22.9,"t1rainra":0,"t1raindy":0,"t1rainhr":0,"t1rainmth":0,"t1rainwy":0,"t1rainyr":0,"t1uvi":0,"t1wdir":276,"t1wgust":0,"t1ws":0,"t1ws10mav":0,"inbat":1,"t1bat":0,"t234c1cn":0,"t234c2cn":0,"t234c3cn":0,"t234c4cn":0,"t234c5cn":0,"t234c6cn":0,"t234c7cn":0,"t11cn":0,"t10cn":0,"t9cn":0,"t6c1cn":0,"t6c2cn":0,"t6c3cn":0,"t6c4cn":0,"t6c5cn":0,"t6c6cn":0,"t6c7cn":0,"t5lscn":0,"t1cn":1,"t8cn":0,"api_ver":"v1.00"}'`